src/table/context/TableContext.tsx

```diff
@@ -52,6 +54,7 @@ export const TableContextProvider = ({
       embed,
       changeSortOrder,
       applyColumnWidths,
+      styling,
```

Comment:
Not sure this should be a base property. We don't really have a definition for a base property but I would consider it to be something that is part of the nebula environment/context and NOT a value that we resolve/create ourself.

So in this example the styling is derived from some of that "nebula environment/context".

Maybe something structure like this would make more sense as to what the different values are:
```js
const value = {
    state: {
      headRowHeight,
      setHeadRowHeight,
      focusedCellCoord,
      setFocusedCellCoord,
      selectionState,
      selectionDispatch: selectionDispatchMock || selectionDispatch,
      hoverIndex,
      setHoverIndex,
    },
    props: {
      changeSortOrder,
      styling,
    },
   env: {
      selectionsAPI,
      layout,
      model,
      translator,
      constraints,
      theme,
      keyboard,
      rootElement,
      embed,
    },
  }
```

Anyway overkill for this PR but something to think about.

Reply:
I see your point. So the props outside of baseProps are carefully put seperately, so you fetch them, they are the same unless its value had changed, so for example:
`const focusedCellCoord = useContextSelector(TableContext, (value) => value.focusedCellCoord);`
That being said if we would put that inside a something called state or similar, I think this should be the same thing:
`const focusedCellCoord = useContextSelector(TableContext, (value) => value.state.focusedCellCoord);`

Reply:
I agree we should seperate them kind of like you suggested, but for now I think we can consider `baseProps` as everything that is kind of updated on layout/model changes or comes straight from index